File: backend/src/main/java/br/ufpr/sept/so2/modules/academico/infrastructure/persistence/PeriodoLetivoJpaEntity.java
package br.ufpr.sept.so2.modules.academico.infrastructure.persistence;

import br.ufpr.sept.so2.modules.academico.domain.PeriodoLetivo;
import br.ufpr.sept.so2.shared.infrastructure.persistence.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import java.time.LocalDate;

@Entity
@Table(name = "periodo_letivo")
public class PeriodoLetivoJpaEntity extends BaseEntity {

    @Column(nullable = false)
    private short ano;

    @Column(nullable = false)
    private short semestre;

    @Column(nullable = false)
    private LocalDate inicio;

    @Column(nullable = false)
    private LocalDate fim;

    @Column(nullable = false)
    private boolean ativo;

    protected PeriodoLetivoJpaEntity() {
    }

    public static PeriodoLetivoJpaEntity fromDomain(PeriodoLetivo periodo) {
        PeriodoLetivoJpaEntity entity = new PeriodoLetivoJpaEntity();
        entity.setId(periodo.getId());
        entity.merge(periodo);
        return entity;
    }

    public void merge(PeriodoLetivo periodo) {
        this.ano = (short) periodo.getAno();
        this.semestre = (short) periodo.getSemestre();
        this.inicio = periodo.getInicio();
        this.fim = periodo.getFim();
        this.ativo = periodo.isAtivo();
    }

    public PeriodoLetivo toDomain() {
        return new PeriodoLetivo(
                getId(),
                ano,
                semestre,
                inicio,
                fim,
                ativo,
                getCreatedAt(),
                getUpdatedAt()
        );
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/academico/infrastructure/persistence/PeriodoLetivoJpaEntity.kt
package br.ufpr.sept.so2.modules.academico.infrastructure.persistence

import br.ufpr.sept.so2.modules.academico.domain.PeriodoLetivo
import br.ufpr.sept.so2.shared.infrastructure.persistence.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import java.time.LocalDate

@Entity
@Table(name = "periodo_letivo")
class PeriodoLetivoJpaEntity : BaseEntity() {
    @Column(nullable = false)
    var ano: Short = 0

    @Column(nullable = false)
    var semestre: Short = 0

    @Column(nullable = false)
    var inicio: LocalDate? = null

    @Column(nullable = false)
    var fim: LocalDate? = null

    @Column(nullable = false)
    var ativo: Boolean = false

    fun merge(periodo: PeriodoLetivo) {
        ano = periodo.ano.toShort()
        semestre = periodo.semestre.toShort()
        inicio = periodo.inicio
        fim = periodo.fim
        ativo = periodo.ativo
    }

    fun toDomain() = PeriodoLetivo(
        requireNotNull(id),
        ano.toInt(),
        semestre.toInt(),
        requireNotNull(inicio),
        requireNotNull(fim),
        ativo,
        requireNotNull(createdAt),
        requireNotNull(updatedAt),
    )

    companion object {
        fun fromDomain(periodo: PeriodoLetivo) = PeriodoLetivoJpaEntity().apply {
            id = periodo.id
            merge(periodo)
        }
    }
}
